app/assets/javascripts/shared/dradis_datatable.js

```diff
@@ -1,16 +1,49 @@
 class DradisDatatable {
   constructor(selector) {
     this.$table = $(selector);
+    if (!this.$table.length) {
+      console.warn('Table not found.');
+      return;
+    }
+
     this.dataTable = null;
+    this.tableHeaders = Array.from(this.$table[0].querySelectorAll('thead th, thead td'));
     this.init();
   }
 
   init() {
+    var colvisColumnIndexes = this.tableHeaders.reduce(function(indexes, column, index) {
```

Comment:
What's the reason for using `reduce` here? According to the MDN docs:

> The reduce() method executes a reducer function (that you provide) on each element of the array, resulting in a single output value.

By my understanding, it looks like we're doing some sort of `map` here to output an array not a single output value. Can we use this instead: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map ? Or `forEach`?

Reply:
I wanted to iterate over an array and filter it based on the `data-colvis="false"` attribute and return the array.

But I think I can probably use standard way.